pkg/session/syssession/pool.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syssession
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/util/intest"
+	"github.com/pingcap/tidb/pkg/util/logutil"
+	"go.uber.org/zap"
+)
+
+// PoolMaxSize is the maximum size of the session pool.
+const PoolMaxSize int = 1024 * 1024 * 1024
+
+// Factory is a function to create a new session context
+type Factory func() (SessionContext, error)
+
+// Pool is an interface for system internal session pool.
+type Pool interface {
+	// Get gets a session from the session pool.
+	Get() (*Session, error)
+	// Put puts the session back to the pool.
+	Put(*Session)
+	// WithSession executes the input function with the session.
+	// After the function called, the session will be returned to the pool automatically.
+	WithSession(func(*Session) error) error
+}
+
+// AdvancedSessionPool is a recyclable resource pool for the system internal session.
+type AdvancedSessionPool struct {
+	noopOwnerHook
+	ctx     context.Context
+	cancel  context.CancelFunc
```

Comment:
Do we need to move the cancel into the mu struct? It seems we try to protect it when we call it in the close function.

Reply:
done